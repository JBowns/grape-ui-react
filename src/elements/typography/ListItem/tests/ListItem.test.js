import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { ListItem } from '..';
import 'jest-styled-components';

describe('ListItem Component base', () => {
  it('should return a ListItem object with base styling', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem>Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('ListItem Component specified styles', () => {
  it('should return a ListItem object with font-style: italic', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem italic>Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with font-size: lg', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem lg>Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with font-size: sm', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem sm>Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a base color (implicit)', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem color="green">Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a base color', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem color="green.base">Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a light color', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem color="green.light">Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a dark color', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem color="green.dark">Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with ellipsis', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem ellipsis>Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a specified font-family', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem fontFamily="monospace">Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a specified font-weight', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem fontWeight="bold">Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a specified kerning', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem kerning="2px">Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a specified text-align', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem textAlign="right">Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a specified text-decoration', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem textDecoration="underline">Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with custom margins', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><ListItem mb={1}>Lorem Ipsum</ListItem></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
