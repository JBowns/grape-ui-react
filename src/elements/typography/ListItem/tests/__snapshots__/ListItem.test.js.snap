// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Component base should return a ListItem object with base styling 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with a base color (implicit) 1`] = `
.c0 {
  color: hsl(122.4,39.4%,49.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with a base color 1`] = `
.c0 {
  color: hsl(122.4,39.4%,49.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with a dark color 1`] = `
.c0 {
  color: hsl(122.4,39.4%,29.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with a light color 1`] = `
.c0 {
  color: hsl(122.4,39.4%,69.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with a specified font-family 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: Consolas,"Andale Mono WT","Andale Mono","Lucida Console","Lucida Sans Typewriter","DejaVu Sans Mono","Bitstream Vera Sans Mono","Liberation Mono","Nimbus Mono L",Monaco,"Courier New",Courier,monospace;
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with a specified font-weight 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: bold;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with a specified kerning 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with a specified text-align 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: right;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with a specified text-decoration 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with custom margins 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 4px;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with ellipsis 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with font-size: lg 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.2rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with font-size: sm 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component specified styles should return a ListItem object with font-style: italic 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  font-style: italic;
  margin-bottom: 0.25rem;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;
